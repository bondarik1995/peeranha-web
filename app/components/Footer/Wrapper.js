import styled from 'styled-components';

const Wrapper = styled.footer`
  position: absolute;
  bottom: 0;
  width: 100%;
  height: 60px;
  background-color: #6c757d;
  color: rgba(255, 255, 255, 0.6);
  a {
    color: #fff !important;
  }
`;

export default Wrapper;
