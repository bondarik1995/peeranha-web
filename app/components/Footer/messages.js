/*
 * Footer Messages
 *
 * This contains all the text for the Footer component.
 */

import { defineMessages } from 'react-intl';

const CURRENT_YEAR = new Date().getFullYear();

export default defineMessages({
  copyrightSymbol: {
    id: 'app.components.Footer.copyrightSymbol',
    defaultMessage: 'Â© ',
  },
  currentYear: {
    id: 'app.components.Footer.currentYear',
    defaultMessage: `${CURRENT_YEAR} `,
  },
  header: {
    id: 'app.components.Footer.header',
    defaultMessage: `Copyright: Peerania`,
  },
});
