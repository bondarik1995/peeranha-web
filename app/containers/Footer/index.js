/**
 *
 * Footer
 *
 */

import React from 'react';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { Link } from 'react-router-dom';

import Wrapper from './styledComponents/wrapper';

/* eslint-disable react/prefer-stateless-function */
export class Footer extends React.Component {
  render() {
    const CURRENT_YEAR = new Date().getFullYear();
    return (
      <Wrapper className="page-footer font-small blue">
        <div className="footer-copyright text-center py-3">
          Â© {CURRENT_YEAR} Copyright:
          <Link to="/" href="/">
            Peerania
          </Link>
        </div>
      </Wrapper>
    );
  }
}

Footer.propTypes = {};

function mapDispatchToProps(dispatch) {
  return {
    dispatch,
  };
}

const withConnect = connect(
  null,
  mapDispatchToProps,
);

export default compose(withConnect)(Footer);
