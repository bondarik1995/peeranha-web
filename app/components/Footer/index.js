/**
 *
 * Footer
 *
 */

import React from 'react';
import { Link } from 'react-router-dom';

import Wrapper from './Wrapper';
import messages from './messages';

const Footer = () => (
  <Wrapper className="page-footer font-small blue">
    <div className="footer-copyright text-center py-3">
      <Link to="/" href="/">
        {messages.copyrightSymbol.defaultMessage}
        {messages.currentYear.defaultMessage}
        {messages.header.defaultMessage}
      </Link>
    </div>
  </Wrapper>
);

export default Footer;
