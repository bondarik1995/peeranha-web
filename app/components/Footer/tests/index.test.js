import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Footer from '../index';

describe('<Footer />', () => {
  it('should match the snapshot', () => {
    const renderedComponent = renderer.create(<Footer />).toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });
});
