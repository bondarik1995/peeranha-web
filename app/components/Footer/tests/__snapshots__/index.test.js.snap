// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should match the snapshot 1`] = `
.c0 {
  position: absolute;
  bottom: 0;
  width: 100%;
  height: 60px;
  background-color: #6c757d;
  color: rgba(255,255,255,0.6);
}

.c0 a {
  color: #fff !important;
}

<footer
  className="page-footer font-small blue c0"
>
  <div
    className="footer-copyright text-center py-3"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <span>
        Â© 
      </span>
      <span>
        2018 
      </span>
      <span>
        Copyright: Peerania
      </span>
    </a>
  </div>
</footer>
`;
